version: "3.8"

services:
  mlflow:
    build: .
    restart: always
    volumes:
      - ./wait-for-it.sh:/wait-for-it.sh
      - ./basic_auth.ini:/usr/local/lib/python3.11/site-packages/mlflow/server/auth/basic_auth.ini
    env_file:
      - .env
    entrypoint: bash ./wait-for-it.sh postgres:5432 -t 90 -- mlflow server --app-name basic-auth --backend-store-uri postgresql://mlflow_user:mlflow_user@postgres:5432/mlflow_db --default-artifact-root s3://mlflow/ --host 0.0.0.0 --port 5000
    ports:
      - "5000:5000"
    depends_on:
      - postgres
    networks:
      - lab

  postgres:
    image: postgres:17.2

    restart: always
    env_file:
      - .env
    volumes:
      - postgres-data:/var/lib/postgresql/data
    networks:
      - lab

  postgres_auth:
    image: postgres:17.2

    restart: always
    env_file:
      - ./auth_db.env
    volumes:
      - postgres-auth-data:/var/lib/postgresql/data
    networks:
      - lab

  minio:
    image: quay.io/minio/minio:RELEASE.2025-01-18T00-31-37Z
    restart: always
    command: server --console-address ":9001"  /data
    ports:
      - 9000:9000
      - 9001:9001
    env_file:
      - ./minio.env
    healthcheck:
      test: ["CMD", "curl", "-f", "http://localhost:9000/minio/health/live"]
      interval: 30s
      timeout: 20s
      retries: 3
    volumes:
      - minio-data:/data
    networks:
      - lab

volumes:
  postgres-data:
  postgres-auth-data:
  minio-data:
  
networks:
  lab:
    external: true
